### DISCUSSION TITLE
Do we know if IDEs (VS/Rider etc) will show Extension Type Properties when we are debugging and inspect an object?

### DISCUSSION THREAD

**Discussion by grofit** [dotnet/csharplang#8254] [Q&A]
As C# 13 is bringing in Extension Types which lets you add properties to classes/interfaces, do we know what the recommended steering is going to be for IDEs when debugging?

For example, if we use the code shown in the C#13 docs for the feature:
```csharp
public implicit extension PersonExtension for Person
{
    public bool IsLead  => this.Organization.Teams.Any(team => team.Lead == this);
}
```

If I were to do `var somePerson = new Person();` then inspect `somePerson ` in the debugger should I see `IsLead` in the properties?

I know this is sort of a question for the IDE creators such as MS for VS and Jetbrains for Rider, but I assume the C# technical team give some steering as to what is their expected behaviour here.

So can anyone provide any insight into if this would be automatically shown or if it would be up to the user to manually watch the `somePerson.IsLead` property? 

(Just to be clear this is not me asking for this feature, I am just wondering if any guidance has been given as to how it should be handled).

**Comment by colejohnson66** [comment]
Interactions with debuggers are part of the runtime, not the language.
